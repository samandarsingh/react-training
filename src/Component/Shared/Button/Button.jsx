import React from "react";
import PropTypes from "prop-types";
import styles from "./Button.module.scss";
/**
 * this is button field component using in shared
 * @param {string} buttonType 
 * @param {string} buttonName
 * @param {string} buttonClass 
 * @param {func} handleChange
 * @param {bool} index
 * @param {bool} status
 * @param {bool} disabled
 * @returns element
 */ 
const Button = props => {
    const {
        buttonType,
        buttonName,
        buttonClass,    
        handleOnClick,
        index,
        status,
        disabled        
    } = props;
    return (        
        <button
            type={buttonType}
            name={buttonName}            
            className={styles[buttonClass]}    
            onClick={e => handleOnClick(e,index,status)}  
            disabled={disabled}               
        >     
            {buttonName}     
        </button>        
    );
}
// set default props
Button.defaultProps = {
    buttonType: "",
    buttonName: "",
    buttonClass: "",    
    handleOnClick:  () => {},
    index: "",
    status: true,
    disabled: false
}
// proptypes validation
Button.propTypes = {
    buttonType: PropTypes.string.isRequired,
    buttonName: PropTypes.string,
    buttonClass: PropTypes.string,    
    handleOnClick: PropTypes.func,
    status: PropTypes.bool,
    disabled: PropTypes.bool.isRequired
  };
export default Button;