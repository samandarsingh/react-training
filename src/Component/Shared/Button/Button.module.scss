.buttonClass {
    padding: 10px 20px;
    background-color: dodgerblue;
    border: 1px solid #ddd;
    color: white;
    cursor: pointer;
}
.success {
    background: rgb(28, 184, 65); /* this is a green */
}
.error {
    background: rgb(202, 60, 60); /* this is a maroon */
}
.warning {
    background: rgb(223, 117, 20); /* this is an orange */
}
.secondary {
    background: rgb(66, 184, 221); /* this is a light blue */
}
.statusButton {
    background-color: white; 
    color: black; 
    border: 2px solid #008CBA;
}
.declineButton {
    background-color: white; 
    color: black; 
    border: 2px solid #f44336;
}
.isCompletedButton {
    background-color: white; 
    color: black; 
    border: 2px solid #4CAF50;
}