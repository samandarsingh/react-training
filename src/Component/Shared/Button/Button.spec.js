import React from "react";
import { mount } from "enzyme";
import Button from "./Button";
describe(`Testing Button component`, () => {
    let wrapper = "";
    beforeAll(() => {
        const props = {
            buttonType: "button",
            buttonName: "",
            buttonClass: "",    
            handleOnClick:  () => {},
            index: "",
            status: true,
            disabled: false
        }
        wrapper = mount(<Button {...props}/>);
    });
    it(`Testing Button component run should be with out crashing`, () => {        
        expect(wrapper).toBeTruthy();
    });
    it(`testing button component getting props`, () => {        
        expect(
            wrapper
            .find(Button)
            .props()
            .buttonType
        ).toEqual(`button`);
    });    
    it(`SnapShot Testing`, () => {
        expect(wrapper).toMatchSnapshot();
    });
});