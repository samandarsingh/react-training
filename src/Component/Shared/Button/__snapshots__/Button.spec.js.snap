// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button component SnapShot Testing 1`] = `ReactWrapper {}`;
